steamclock/steamclog-android
# [#110] Added UserReports method; and support for adding log and extra data

### Related Issue: #110 


### Summary of Problem:
* Android did not yet include the `UserReport` method
* On top of that, we'd like to be able to automatically add the detailed log file to a user report, as well as some extra diagnostic data

### Proposed Solution:
* Added UserReport method, which fires off a Timber error log
* Added `detailedLogsOnUserReports` config; if true the "latest" log file will be attached to all UserReports
* Added `extraInfo` config; this allows a key/value mapping to be set for a report purpose 
* Now using Sentry's `captureEvent` instead of `captureMessage`
* Updated sample app to provide ways to test these new features

### Testing Steps:
1. Go to the Sentry page for the sample app: https://steamclock-software.sentry.io/issues/?project=5399932
2. Select all of the issues in the list and select the "Resolve" button; this should clear the list of errors and make it easy to see new ones coming in.
3. Run the sample app, and change the log level at the top to Release
4. Press the "Report Single Error" button (wait 10ish seconds for Sentry to be notified, refresh Sentry issues page)
5. Make sure you only see 1 error report, click into it
6. Scroll down and make sure the extra details have been added and NO error log is attached:
![image](https://github.com/steamclock/steamclog-android/assets/5217641/474e8cc9-e1f8-4cac-b9ac-d53253d17219)
7. Go back to the sample app, make sure "Add logs on User Reports" is checked and click the "Send User Report" button
8. Wait 10ish seconds for Sentry to be notified, refresh Sentry issues page and make sure "This is my User Report" now shows, click into it
9. Scroll down and make sure the extra details AND the log attachment are present:
![image](https://github.com/steamclock/steamclog-android/assets/5217641/33334adc-11fe-4c2d-beb6-9acbbac5dd94)
10. Go back to the sample app, disable "Add logs on User Reports" and send another User report.
11. Make sure it appears in Sentry, click into it make sure the extra details are there but the log it NOT attached.
